src/query/plan/operator.cpp

```diff
@@ -3007,58 +3007,71 @@ bool Delete::DeleteCursor::Pull(Frame &frame, ExecutionContext &context) {
   OOMExceptionEnabler oom_exception;
   SCOPED_PROFILE_OP("Delete");
 
-  if (delete_executed_) {
-    return false;
+  if (self_.buffer_size_ && buffer_size_ == -1) {
+    // Limit expression doesn't contain identifiers so graph view is not
+    // important.
+    ExpressionEvaluator evaluator(&frame, context.symbol_table, context.evaluation_context, context.db_accessor,
+                                  storage::View::OLD);
+    TypedValue buffer_size = self_.buffer_size_->Accept(evaluator);
+    if (buffer_size.type() != TypedValue::Type::Int)
+      throw QueryRuntimeException("Number of periodically deleted elements must be an integer.");
+
```

Comment:
Can we cover this part in planning so we don't have to worry about that here? I would put here DMG_ASSERT that this is int type

Reply:
covered in separate function. Unfortunately I can't use the expression evaluator here, as that is some other scope (making sure expression eval works with planning and not only execution)